yarn-project/pxe/src/synchronizer/synchronizer.ts

```diff
@@ -42,17 +43,19 @@ export class Synchronizer {
     }
     this.running = true;
 
-    await this.initialSync();
+    await this.jobQueue.put(this.initialSync);
 
     const run = async () => {
       while (this.running) {
-        if (this.noteProcessorsToCatchUp.length > 0) {
-          // There is a note processor that needs to catch up. We hijack the main loop to catch up the note processor.
-          await this.workNoteProcessorCatchUp(limit, retryInterval);
-        } else {
-          // No note processor needs to catch up. We continue with the normal flow.
-          await this.work(limit, retryInterval);
-        }
+        await this.jobQueue.put(async () => {
```

Comment:
We will need to change this sightly. Currently, the `work` functions sleep if there is nothing more to do.

So we want the `work` functions to return indicating if there is more work to do. Then inside the job we loop until there is no more work to be done. Finally, at the outer layer (outside of the job queue) we sleep for the `retryInterval`. 

Does that sound like it works?